ui/packages/consul-ui/app/components/data-writer/index.js

```diff
@@ -21,5 +21,17 @@ export default Component.extend(Slotted, {
       set(this, 'data', data);
       this.dispatch('PERSIST');
     },
+    error: function(data, e) {
+      if (e && typeof e.preventDefault === 'function') {
```

Comment:
why are we `preventingDefault` here? can we add a test-case for this change?

Reply:
If the event that is fired has some sort of default we want to prevent it.

This is currently a `noop`, as I'm pretty sure events passed here will never have a default. So happy to leave in or take out, it mainly for consistency with the above `persist`. We might eventually have a case where the event does have a default that needs preventing, and this way both `persist` and `error` work the same way. 

All in all I'm easy either way.

> can we add a test-case for this change?

Which bit would you like me to test here? The

```javascript
e && typeof e.preventDefault === 'function'
```
?

Reply:
a test case testing the newly added behavior would be great

Reply:
Can I just make sure you understand what was changed here?

We didn't add any new behaviour as such, we just added a fork for a different shape of data. So to me right now, it sounds like you want me to test either (or both?)

```javascript
e && typeof e.preventDefault === 'function'
```

and/or

```javascript
typeof data.error.errors !== 'undefined'
```

Just to note, acceptance tests for the whole peering feature will be added once everything is in.

Reply:
Is there a test for this behavior anywhere? You are changing code but you are not adding/updating a test-case for the change. We are also adding behavior of `preventDefault`ing the regular event. Why is this needed? Do we want to add a test-case that showcases why this change is necessary?

We can add that as a follow-up as the change is probably fine but don't we want to add a test harness for this behavior?